start.js

```diff
@@ -42,19 +42,22 @@ const trim = (value, charlist) => trimLeft(trimRight(value, charlist));
 const main = async () => {
     let branch = process.env.INPUT_BRANCH;
     const repository = trim(process.env.INPUT_REPOSITORY || process.env.GITHUB_REPOSITORY);
+    const github_url = trim(process.env.INPUT_GITHUB_URL)
     if (!branch) {
         const headers = {
             'User-Agent': 'github.com/ad-m/github-push-action'
         };
         if (process.env.INPUT_GITHUB_TOKEN) headers.Authorization = `token ${process.env.INPUT_GITHUB_TOKEN}`;
-        const body = JSON.parse(await get(`https://api.github.com/repos/${repository}`, { headers }))
+        const api_url = github_url === 'github.com' ? 'api.github.com' : github_url + '/api/v3';
```

Comment:
@ad-m This is the source of the bug

Reply:
Thanks, for the hint. I've opened a fresh modified draft PR.